pkg/mlsvalidate/service.go

```diff
@@ -16,42 +17,31 @@ import (
 	envelopesProto "github.com/xmtp/xmtpd/pkg/proto/xmtpv4/envelopes"
 	"github.com/xmtp/xmtpd/pkg/utils"
 	"go.uber.org/zap"
-	"google.golang.org/grpc"
 )
 
 type MLSValidationServiceImpl struct {
 	grpcClient svc.ValidationApiClient
 }
 
-func NewMlsValidationService(
+var _ MLSValidationService = (*MLSValidationServiceImpl)(nil)
+
+func NewMLSValidationService(
 	ctx context.Context,
 	logger *zap.Logger,
 	cfg config.MlsValidationOptions,
 	clientMetrics *grpcprom.ClientMetrics,
 ) (*MLSValidationServiceImpl, error) {
-	target, isTLS, err := utils.HTTPAddressToGRPCTarget(cfg.GrpcAddress)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert HTTP address to gRPC target: %w", err)
-	}
-
-	creds, err := utils.GetCredentialsForAddress(isTLS)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get credentials: %w", err)
-	}
-
 	logger.Info(
 		"connecting to mls validation service",
 		zap.String("url", cfg.GrpcAddress),
-		zap.String("target", target),
 	)
-	conn, err := grpc.NewClient(
-		target,
-		grpc.WithTransportCredentials(creds),
```

Comment:
what handles TLS now?

Reply:
TLS is handled in `pkg/utils/api_clients.go:NewGRPCConn()`.
We use TLS when: gateway connects to a node, node to node, node to MLS validation service. `NewGRPCConn` handles the complexities.